pkg/test/framework/components/istio/kubealpha.go

```diff
@@ -0,0 +1,241 @@
+	for _, ns := range namespaces {
+		ns := ns
+		wg.Add(1)
+		go func() {
+			err := i.environment.Accessor.DeleteNamespace(ns)
+
+			if err == nil {
+				err = i.environment.Accessor.WaitForNamespaceDeletion(ns)
+			}
+
+			if err != nil {
+				errchan <- err
+			}
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+
+	close(errchan)
+	var err error
+	for e := range errchan {
+		err = multierror.Append(err, e)
+	}
+
+	return err
+}
+
+func (i *installerComponent) Dump() {
+	scopes.CI.Errorf("=== Dumping Istio Deployment State...")
+
+	d, err := i.ctx.CreateTmpDirectory("istio-state")
+	if err != nil {
+		scopes.CI.Errorf("Unable to create directory for dumping Istio contents: %v", err)
+		return
+	}
+	for _, ns := range GetDeployedNamespaces(i.settings) {
+		deployment.DumpPodState(d, ns, i.environment.Accessor)
+		deployment.DumpPodEvents(d, ns, i.environment.Accessor)
+
+		pods, err := i.environment.Accessor.GetPods(ns)
+		if err != nil {
+			scopes.CI.Errorf("Unable to get pods from the system namespace: %v", err)
+			return
+		}
+
+		for _, pod := range pods {
+			for _, container := range pod.Spec.Containers {
+				l, err := i.environment.Logs(pod.Namespace, pod.Name, container.Name, false /* previousLog */)
+				if err != nil {
+					scopes.CI.Errorf("Unable to get logs for pod/container: %s/%s/%s", pod.Namespace, pod.Name, container.Name)
+					continue
+				}
+
+				fname := path.Join(d, fmt.Sprintf("%s-%s.log", pod.Name, container.Name))
+				if err = ioutil.WriteFile(fname, []byte(l), os.ModePerm); err != nil {
+					scopes.CI.Errorf("Unable to write logs for pod/container: %s/%s/%s", pod.Namespace, pod.Name, container.Name)
+				}
+			}
+		}
+	}
+}
+
+func deployAlphaInstall(ctx resource.Context, env *kube.Environment, cfg Config) (Instance, error) {
+	scopes.CI.Infof("=== Istio Component Config ===")
+	scopes.CI.Infof("\n%s", cfg.String())
+	scopes.CI.Infof("================================")
+
+	i := &installerComponent{
+		environment: env,
+		settings:    cfg,
+		ctx:         ctx,
+	}
+	i.id = ctx.TrackResource(i)
+
+	if !cfg.DeployIstio {
+		scopes.Framework.Info("skipping deployment due to Config")
+		return i, nil
+	}
+
+	// Top-level work dir for Istio deployment.
+	workDir, err := ctx.CreateTmpDirectory("istio-deployment")
+	if err != nil {
+		return nil, err
+	}
+
+	// Create helm working dir
+	helmWorkDir := path.Join(workDir, "helm")
+	if err := os.MkdirAll(helmWorkDir, os.ModePerm); err != nil {
+		return nil, err
+	}
+
+	// First, generate CRDs.
+	crdYaml, err := generateCRDYaml(path.Join(ienv.InstallDir, "crds/files"))
+	if err != nil {
+		return nil, err
+	}
+
+	scopes.CI.Infof("Applying CRDs: %v", workDir)
+
+	// Apply CRDs first.
+	if _, err = env.Accessor.ApplyContents("", crdYaml); err != nil {
+		return nil, err
+	}
+
+	// Restructure components for easier lookup
+	components := map[InstallComponent]struct{}{}
+	for _, c := range cfg.InstallComponents {
+		components[c] = struct{}{}
+	}
+
+	if _, f := components[Base]; f {
```

Comment:
I think that's usually called "ok". Though if you need this kind of tests it may make more sense to simply make it a map of bools instead of struct{}.